sim/paladin/retribution_talents.go

```diff
@@ -0,0 +1,237 @@
+package paladin
+
+import (
+	"github.com/wowsims/cata/sim/core"
+	"time"
+)
+
+func (paladin *Paladin) applyRetributionTalents() {
+	paladin.applyCrusade()
+	paladin.applyRuleOfLaw()
+	paladin.applySanctityOfBattle()
+	paladin.applySealsOfCommand()
+	paladin.applySanctifiedWrath()
+	paladin.applyCommunion()
+	paladin.applyArtOfWar()
+	paladin.applyDivinePurpose()
+	paladin.applyInquiryOfFaith()
+}
+
+func (paladin *Paladin) applyCrusade() {
+	if paladin.Talents.Crusade == 0 {
+		return
+	}
+
+	paladin.AddStaticMod(core.SpellModConfig{
+		ClassMask:  SpellMaskCrusaderStrike | SpellMaskDivineStorm | SpellMaskTemplarsVerdict | SpellMaskHolyShock,
+		Kind:       core.SpellMod_DamageDone_Flat,
+		FloatValue: 0.1 * float64(paladin.Talents.Crusade),
+	})
+
+	// TODO: Add Healing Mod for Holy Shock if healing sim gets implemented
+}
+
+func (paladin *Paladin) applyRuleOfLaw() {
+	if paladin.Talents.RuleOfLaw == 0 {
+		return
+	}
+
+	paladin.AddStaticMod(core.SpellModConfig{
+		ClassMask:  SpellMaskCrusaderStrike | SpellMaskWordOfGlory | SpellMaskHammerOfTheRighteous,
+		Kind:       core.SpellMod_BonusCrit_Rating,
+		FloatValue: 5 * float64(paladin.Talents.RuleOfLaw) * core.CritRatingPerCritChance,
+	})
+}
+
+func (paladin *Paladin) applySanctityOfBattle() {
+	if !paladin.Talents.SanctityOfBattle {
+		return
+	}
+
+	spenderCooldownMod := paladin.AddDynamicMod(core.SpellModConfig{
+		Kind:      core.SpellMod_Cooldown_Flat,
+		ClassMask: SpellMaskCrusaderStrike | SpellMaskDivineStorm,
+	})
+
+	paladin.AddOnCastSpeedChanged(func(_ float64, castSpeed float64) {
+		spenderCooldownMod.UpdateTimeValue(-(4500*time.Millisecond - time.Duration(4500*castSpeed)))
+	})
+
+	core.MakePermanent(paladin.GetOrRegisterAura(core.Aura{
+		Label:      "Sanctity of Battle",
+		ActionID:   core.ActionID{SpellID: 25956},
+		BuildPhase: core.CharacterBuildPhaseAll,
```

Comment:
remove this build phase. This is only used to show stats in the stats panel and for nothing else.